snakemake_executor_plugin_latch/__init__.py

```diff
@@ -0,0 +1,255 @@
+                "argGpuLimit": gpus,
+                "argGpuType": gpu_type,
+                "argParentJobIds": list(upstream),
+            },
+        )
+
+        self.report_job_submission(SubmittedJobInfo(job))
+
+    async def check_active_jobs(
+        self, active_jobs: List[SubmittedJobInfo]
+    ) -> AsyncGenerator[SubmittedJobInfo, None]:
+
+        # Check the status of active jobs.
+
+        # You have to iterate over the given list active_jobs.
+        # If you provided it above, each will have its external_jobid set according
+        # to the information you provided at submission time.
+        # For jobs that have finished successfully, you have to call
+        # self.report_job_success(active_job).
+        # For jobs that have errored, you have to call
+        # self.report_job_error(active_job).
+        # This will also take care of providing a proper error message.
+        # Usually there is no need to perform additional logging here.
+        # Jobs that are still running have to be yielded.
+        #
+        # For queries to the remote middleware, please use
+        # self.status_rate_limiter like this:
+        #
+        # async with self.status_rate_limiter:
+        #    # query remote middleware here
+        #
+        # To modify the time until the next call of this method,
+        # you can set self.next_sleep_seconds here.
+
+        job_infos_by_id = {x.job.jobid: x for x in active_jobs}
+
+        statuses = (
+            await self.async_gql_session.execute_async(
+                gql.gql(
+                    """
+                    query CheckActiveJobs($argExecutionToken: String!, $argJobIds: [BigInt!]!) {
+                        smMultiGetJobInfos(
+                            argExecutionToken: $argExecutionToken
+                            argJobIds: $argJobIds
+                        ) {
+                            nodes {
+                                jobId
+                                status
+                                errorMessage
+                            }
+                        }
+                    }
+                    """
+                ),
+                {
+                    "argExecutionToken": self.execution_token,
+                    "argJobIds": [x.job.jobid for x in active_jobs],
+                },
+            )
+        )["smMultiGetJobInfos"]
+
+        for node in statuses["nodes"]:
+            status = node["status"]
+            job_info = job_infos_by_id[int(node["jobId"])]
+
+            if status in {"SUCCEEDED", "SKIPPED"}:
+                self.report_job_success(job_info)
+            elif status in {"FAILED", "ABORTED"}:
+                self.report_job_error(job_info, node["errorMessage"])
+            else:
+                yield job_info
+
+    def cancel_jobs(self, active_jobs: List[SubmittedJobInfo]):
+        # Cancel all active jobs.
+        # This method is called when Snakemake is interrupted.
```

Comment:
maybe add a comment here saying that garabge collector handles task cleanup. 
if you want the abort to be instant (which I would reccommend, leads to better UX), you can do what Nextflow does where we set the status to ABORTING and then kill the pod in dispatcher